### PR TITLE
fix: Get numeric value for Enum fields if it is configured as query param or path param

### PR DESCRIPTION
**Pull request by blakeli0**
Get numeric value for Enum fields if it is configured as query param or path param. Refactor HttpBinding to include Field info and use builder.
Found an existing issue regarding how query params are populated. Add comments to explain the current behavior and expected behavior for populating query params. Created issue https://github.com/googleapis/gapic-generator-java/issues/1041 to track the future work.

### CODE REVIEW

**Review by diego92sigma6 — APPROVED** [review]

File: src/test/java/com/google/api/generator/gapic/model/HttpBindingsTest.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+  }
+
+  @Test
+  public void isOptional_shouldReturnFalseIfFieldIsNull() {
+    HttpBinding httpBinding = httpBindingBuilder.build();
+    Truth.assertThat(httpBinding.isOptional()).isFalse();
+  }
+
+  @Test
+  public void isOptional_shouldReturnFalseIfFieldExistAndIsOptionalIsFalse() {
```
Comment (alicejli) [inline src/test/java/com/google/api/generator/gapic/model/HttpBindingsTest.java:41]: nit: I think it should read "IfFieldExists" to keep present tense (and elsewhere in tests)

File: src/test/proto/compliance.proto [hunk 2]
```diff
@@ -80,6 +80,44 @@ service Compliance {
       get: "/v1beta1/repeat/{info.f_string=first/*}/{info.f_child.f_string=second/**}:pathtrailingresource"
     };
   }
+
+  // This method requests an enum value from the server. Depending on the contents of EnumRequest, the enum value returned will be a known enum declared in the
```
Comment (alicejli) [inline src/test/proto/compliance.proto:84]: general question - is there a reason this proto doesn't match https://github.com/googleapis/gapic-showcase/blob/main/schema/google/showcase/v1beta1/compliance.proto exactly?

**Review by alicejli — COMMENTED** [review]
Some small nits and a general question on the `compliance.proto`. Otherwise it looks good; thank you for the thorough comments and your hard work on this feature!

File: src/main/java/com/google/api/generator/gapic/model/HttpBindings.java [hunk 3]
```diff
@@ -36,21 +36,54 @@ public enum HttpVerb {
 
   @AutoValue
   public abstract static class HttpBinding implements Comparable<HttpBinding> {
+
+    // The full qualified name of the field. e.g. request.complex_object.another_object.name
```
[the remark this answers is not in the document]
Comment (blakeli0) [reply src/main/java/com/google/api/generator/gapic/model/HttpBindings.java]: Updated

File: src/test/java/com/google/api/generator/gapic/model/HttpBindingsTest.java (hunk 1 above)
Comment (blakeli0) [reply src/test/java/com/google/api/generator/gapic/model/HttpBindingsTest.java]: Updated

File: src/test/proto/compliance.proto (hunk 2 above)
Comment (blakeli0) [reply src/test/proto/compliance.proto:84]: This was way before I joined, but I guess what happened was that the showcase protos have a lot of edge cases that we want to use, so instead of creating protos from scratch for our test cases, we just copied over these showcase protos and use them as they are. However, protos in showcase evolve over time and we didn't update them regularly.
